### PR TITLE
Nova v4

### PR DESCRIPTION
**Pull request by HammenWS**

### CODE REVIEW

**Comment by HammenWS** [comment]
Todo: add upgrade guide in readme.

File: src/PagesTool.php [hunk 1]
```diff
@@ -15,16 +17,20 @@ class PagesTool extends Tool
      */
     public function boot()
     {
-        Nova::resources([PageResource::class]);
+        // Uncomment the following lines to add scripts and styles to the tool
+        // Nova::script('nova-netlify-deploy', __DIR__ . '/../dist/js/tool.js');
+        // Nova::style('nova-netlify-deploy', __DIR__ . '/../dist/css/tool.css');
```
Comment (harmenjanssen) [inline src/PagesTool.php:22]: Hey, hier staat `nova-netlify-deploy`, dat is een andere package. 🕵️

File: src/Resources/PageResource.php [hunk 2]
```diff
@@ -283,10 +277,10 @@ public static function getPageOptionsForSelect(
             })
             ->toArray();
     }
-
     protected function basicFields(): array
     {
         $basicFields = [
+            ID::make()->sortable(),
```
Comment (harmenjanssen) [inline src/Resources/PageResource.php:283]: Why the ID?

File: src/Resources/PageResource.php [hunk 3]
```diff
@@ -350,21 +344,25 @@ protected function basicFields(): array
...
+        // }
         return $basicFields;
     }
 
     protected function templateFields(): array
     {
-        return collect([
-            Select::make(__('pages::pages.fields.template'), 'template')
+        $templateFields = collect([
+            Select::make('Template', 'template')
```
Comment (harmenjanssen) [inline src/Resources/PageResource.php:365]: Why did you remove the translation of the label?

File: src/Resources/PageResource.php [hunk 4]
```diff
@@ -433,33 +432,57 @@ protected static function afterValidation(NovaRequest $request, $validator)
...
 
-    /**
-     * Create NovaDependencyContainers for every template that offers
-     * dependent fields.
-     * These will become visible in the form after switching templates.
-     */
     protected function getTemplateDependentFields(): array
     {
-        return collect(config('nova-pages-tool.templates'))
+        // $fields = [];
```
Comment (harmenjanssen) [inline src/Resources/PageResource.php:437]:
```suggestion
```

File: src/Resources/PageResource.php [hunk 5]
```diff
@@ -433,33 +432,57 @@ protected static function afterValidation(NovaRequest $request, $validator)
...
+    protected function makeTemplateDependentField(
+        Field $field,
+        string $templateName
+    ) {
+        // TODO: Check if fields got the SupportsDependentFields trait
+
+        // Make every field dependent on the template by hiding and showing
+        // https://github.com/laravel/nova-issues/issues/4170
+        $field
+            ->onlyOnForms() // Because depends on logic will not be applied on detail view
```
Comment (harmenjanssen) [inline src/Resources/PageResource.php:476]: Won't this make the field hidden on detail views?

**Review by harmenjanssen — COMMENTED** [review]
Looks good! But I see quite some TODO's still.
Let's polish it a bit more. Also some inline comments, see below.

**Comment by harmenjanssen** [comment]
By the way, do we need the whole front-end scaffolding in this tool, since we don't use it?

**Comment by HammenWS** [comment]
After some testing I noticed that the the new Nova 4 feature of dependent fields won't work on all fields type, like images. For that reason we cannot replace the Dependency Container package with the Nova 4 feature, since we need it for all Field types.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -71,8 +73,8 @@ jobs:
 
       - uses: actions/setup-node@v3
         with:
-          node-version: "12.x"
+          node-version: "16.x"
 
-      - run: npm install
+      - run: yarn
```
Comment (martijngastkemper) [inline .github/workflows/ci.yml:78]:
```suggestion
      - run: yarn install
```

It does exactly the same, but it is more descriptive.

**Review by martijngastkemper — APPROVED** [review]
LGTM!

**Comment by HammenWS** [comment]
Closing since upgrade to v4 is already done via https://github.com/grrr-amsterdam/nova-pages-tool/pull/33
